### PR TITLE
Add support for ATmega128A

### PR DESCRIPTION
**Pull request by DaanDekoningKrekels**
This commit adds basic support for the [ATmega128A](https://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-8151-8-bit-AVR-ATmega128A_Datasheet.pdf). 

I hope the patches are applied as expected.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+- Support for `ATmega128A`.
+
 ### Added
+
```
Comment (Rahix) [inline CHANGELOG.md:12]: Just for the future, in my projects you don't have to update the CHANGELOG as part of your PRs.  I'll take care of this after merging.  Other projects are doing this differently, though - this is usually documented somewhere in the contributing guidelines of each project (if such guidelines exist).

File: patch/atmega128a.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  TWSR:
+    _modify:
+      TWS:
+        access: read-only
+    TWPS:
+      _replace_enum:
+        PRESCALER_1: [0, "Prescaler Value 1"]
+        PRESCALER_4: [1, "Prescaler Value 4"]
+        PRESCALER_16: [2, "Prescaler Value 16"]
+        PRESCALER_64: [3, "Prescaler Value 64"]
```
Comment (Rahix) [inline patch/atmega128a.yaml:35]: You had mentioned some problems with USART and TWI.  I suppose these lines here are your custom patches for those peripherals to fix the problems you talked about?

Comment (DaanDekoningKrekels) [reply patch/atmega128a.yaml:35]: Yes, these lines fixed the errors mentioned here: https://github.com/DaanDekoningKrekels/avr-device/commit/cf4ee34701f95449c5954c9bc3d710e14bbac146#commitcomment-101614870 . 
Right now, it does not give any errors, but I am not yet 100% sure about the correctness...

File: CHANGELOG.md (hunk 1 above)
Comment (DaanDekoningKrekels) [reply CHANGELOG.md:12]: Ohw, I will take it into account!

File: patch/atmega128a.yaml (hunk 2 above)
Comment (Rahix) [reply patch/atmega128a.yaml:35]: Ah, didn't see that.  Changes look good to me.  I'll merge this as-is and we can always go back and fix more things later on.

**Review by Rahix — APPROVED** [review]

**Pull request merged**